File: Services/Teacher/Teacher.Infrastructure/Repositories/CourseRepository.vb
Imports Teacher.Core.Entities
Imports Teacher.Core.Repositories
Imports Teacher.Infrastructure.Extensions
Imports Dapper
Imports Teacher.Core

Public Class CourseRepository
    Implements ICourseRepository

    Public Async Function GetAll() As Task(Of IEnumerable(Of Course)) Implements ICourseRepository.GetAll
        Using connection = Await DapperConnectionProvider.ConnectionAsync()
            Dim courses = Await connection.GetListAsync (Of Course)()
            Return courses
        End Using
    End Function

    Public Async Function GetById(id As String) As Task(Of Course) Implements ICourseRepository.GetById
        Using connection = Await DapperConnectionProvider.ConnectionAsync()
            Return Await connection.GetAsync (Of Course)(id)
        End Using
    End Function
End Class

File: Services/Student/Student.Infrastructure/Repositories/CourseRepository.cs
using Student.Core.Entities;
using Student.Core.Repositories;
using Student.Infrastructure.Extensions;
using Dapper;

namespace Student.Infrastructure.Repositories;

public class CourseRepository : ICourseRepository
{
    public async Task<IEnumerable<Course>> GetAll()
    {
        await using var connection = await DapperConnectionProvider.ConnectionAsync();
        var course = await connection.GetListAsync<Course>();
        return await connection.GetListAsync<Course>();
    }
    public async Task<Course> Get(string id)
    {
        await using var connection = await DapperConnectionProvider.ConnectionAsync();
        return connection.Get<Course>(id);
    }
}
